docs/high/datatype.rst

```diff
@@ -0,0 +1,38 @@
+.. _datatype:
+
+Datatypes
+=========
+
+Usually, :class:`Datatype` objects are created for you on the fly and you do not
+have to worry about them. However, in certain cases, namely when no HDF5
+equivalent exists for a given :class:`dtype`, you must register the :class:`dtype`
+manually for use with h5py.::
+
+    arr = np.array([np.datetime64('2019-06-30')])
+    h5py.register_dtype(arr.dtype)
+    dset = f.create_dataset("datetimes", data=arr)
```

Comment:
I think adding some commentary about the tradeoffs/assumptions when using opaque types to store data (e.g. harder to read vs exact representation of type). My main concern is we're going to get lots of bugs filed from people who assumed that using `h5py.register_dtype` absolves them about thinking how their encoding their data (c.f. pickle being used for long term storage).